AWSCognitoIdentityProvider/AWSCognitoIdentityUser.m

```diff
@@ -203,13 +229,28 @@ -(instancetype) initWithUsername: (NSString *)username pool:(AWSCognitoIdentityU
         self.confirmedStatus = AWSCognitoIdentityUserStatusConfirmed;
 
         NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserAccessToken];
+        NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserIdToken];
+        
+        NSString * idToken = self.pool.keychain[idTokenKey];
         NSString * accessToken = self.pool.keychain[accessTokenKey];
-        //if the session expires > 5 minutes return it and there is at least an accessToken.
-        if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending && accessToken){
-            NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserIdToken];
-            AWSCognitoIdentityUserSession * session = [[AWSCognitoIdentityUserSession alloc] initWithIdToken:self.pool.keychain[idTokenKey] accessToken:accessToken refreshToken:refreshToken expirationTime:expiration];
         
-            session.expirationTime = expiration;
+        AWSCognitoIdentityUserSession * session;
+        
+        // Session is available if we have expiration and accessToken.
+        if (expiration && accessToken) {
+            session = [[AWSCognitoIdentityUserSession alloc] initWithIdToken:idToken
+                                                                 accessToken:accessToken
+                                                                refreshToken:refreshToken
+                                                              expirationTime:expiration];
```

Comment:
We used to set the `session.expirationTime` explicitly after initializing the new session, but now we don't. I assume that was redundant?

Reply:
Yes, the init is doing the same. I am not sure why we had it again, so removed.